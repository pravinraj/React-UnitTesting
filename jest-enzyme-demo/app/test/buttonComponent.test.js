import ButtonComponent from '../buttonComponent.jsx';
import React from 'react';
import { shallow,configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

describe('<ButtonComponent/>',()=>{
	let wrapper;
	beforeEach(()=>{
		wrapper = shallow(<ButtonComponent/>);
	});

	it('Should render and verify snapshot testing of Button component sucessfully',()=>{
		const tree = renderer.create(<ButtonComponent />).toJSON();
  		expect(tree).toMatchSnapshot();
	});
});