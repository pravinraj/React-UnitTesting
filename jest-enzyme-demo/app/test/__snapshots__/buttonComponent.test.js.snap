// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonComponent/> Should render and verify snapshot testing of Button component sucessfully 1`] = `
<div>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Primary
  </button>
  <div>
    Click count test: 
    0
  </div>
</div>
`;
